ResearchSuite/ResearchSuite/RSDFormStepDataSource.swift

```diff
@@ -0,0 +1,221 @@
+//
+//  RSDFormStepDataSource.swift
+//  ResearchSuite
+//
+//  Copyright © 2017 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/**
+ RSDFormStepDataSource: the internal model for a table view controller. It provides the UITableViewDataSource,
+ manages and stores answers provided thru user input, and provides an RSDResult with those anwers upon request.
+ 
+ It also provides several convenience methods for saving or selecting answers, checking if all answers are valid,
+ and retrieving specific model objects that may be needed by the ViewController.
+ 
+ The tableView data source is comprised of 3 objects:
+ 
+ 1) RSDTableSection - An object representing a section in the tableView. It has one or more RSDTableItemGroup objects.
+ 
+ 2) RSDTableItemGroup - An object representing a specific question supplied by RSDStep as an input field.  Upon init(), the ItemGroup will create one or more RSDTableItem objects representing the answer options for the RSDInputField. The ItemGroup is responsible for storing/computing the answers for its RSDInputField.
```

Comment:
I can try... :)

So... when displaying a tableview of cells with input fields representing the cells, in the case of list-style selection, the tableview uses multiple cells in a section to define the list of selection choices for a single input field. This is part of Josh's architecture that I kinda like (and didn't have a better idea for representing). It's designed around the idea that an item group represents a single answer, but may use multiple cells to *get* that answer. In this case, the only items defined at the base level are choice selection but it could be extended to support additional types.  For example:

<img width="482" alt="screen shot 2017-10-27 at 2 22 57 pm" src="https://user-images.githubusercontent.com/5649217/32125794-66ae9d40-bb22-11e7-9e97-eca6997bcce7.png">

This survey question can be represented using two independent text fields for hours and minutes, with custom text for the "skip" action which is displayed as an underlined button at the bottom of the view.

With the itemGroup, this could represent as three cells where the first two cells are text field entry for hours/minutes and the third cell is a choice selection cell with a value == 0.  OR it could be a picker for hours/minutes and a choice selection cell for no exercise.